src/renderer/components/ScreenSharePicker.tsx

```diff
@@ -380,35 +444,83 @@ function ModalComponent({
-                                    height
-                                }
-                            });
+                        try {
+                            var conn = [...MediaEngineStore.getMediaEngine().connections].find(
+                                connection => connection.streamUserId === UserStore.getCurrentUser().id
+                            );
+                        } catch {
+                            console.log("No current stream.");
+                        }
+                        try {
+                            if (conn) {
+                                const track = conn.input.stream.getVideoTracks()[0];
+                                const frameRate = Number(settings.fps);
+                                const height = Number(settings.resolution);
+                                const width = Math.round(height * (16 / 9));
+                                var constraints = track.getConstraints();
+                                const newConstraints = {
+                                    ...constraints,
+                                    frameRate,
+                                    width: { min: 640, ideal: width, max: 2560 },
+                                    height: { min: 480, ideal: height, max: 1440 },
+                                    advanced: [{ width: width, height: height }],
+                                    resizeMode: "none"
+                                };
+                                track.applyConstraints(newConstraints).then(() => {
+                                    console.log("Applied constraints from ScreenSharePicker successfully.");
+                                    console.log("New constraints:", track.getConstraints());
+                                });
+
+                                // changing stream quality description
+                                conn.videoStreamParameters[0].maxFrameRate = Number(settings.fps);
+                                conn.videoStreamParameters[0].maxResolution.height = Number(settings.resolution);
+                                conn.videoStreamParameters[0].maxResolution.width = Math.round(height * (16 / 9));
+                            }
+                        } catch {
+                            console.log("No current stream.");
                         }
+                        try {
+                            submit({
+                                id: selected!,
+                                ...settings
+                            });
 
-                        submit({
-                            id: selected!,
-                            ...settings
-                        });
+                            // reapply contraints after some time to let discord resubmit stream
+                            // i believe there MUST be way to do it cleaner..
+                            if (conn) {
+                                setTimeout(() => {
```

Comment:
avoid weird hacks like this. why is this necessary in the first place?

Reply:
About `setTimeout`: we use it because after `submit()` function work (even using `await` with `then`) discord take some time to set their default screenres x 30 fps settings. If we change constraints without timeout, discord just will set up 30 fps for it instead of picked user settings
I really wish we could do this without `setTimeout()` :c